sqlmesh/core/context.py

```diff
@@ -331,6 +331,23 @@ def __init__(
 
         self.path, self.config = t.cast(t.Tuple[Path, C], next(iter(self.configs.items())))
 
+        # Separate loader and configurations for different project types (dbt, native, hybrid)
+        self._dbt_loader: t.Optional[Loader] = None
+        self._sqlmesh_loader: t.Optional[Loader] = None
+        self.dbt_configs: t.Dict[Path, C] = {}
+        self.sqlmesh_configs: t.Dict[Path, C] = {}
+        for path, config in self.configs.items():
+            project_type = config.loader.PROJECT_TYPE
```

Comment:
Another way to go about this is:

```python
project_type = c.DBT if issubclass(config.loader, DbtLoader) else c.NATIVE
```

So you won't have to define a new constant in the loader. The reason [you can't do](https://github.com/TobikoData/sqlmesh/pull/3122#discussion_r1757654453) `isinstance` is because `config.loader` is a loader _class_, not an object.

Reply:
Yes altering to use `issubclass` it really seems like the more elegant solution for this, thanks!